src/utils/SuggestedCollabs.js

```diff
@@ -0,0 +1,92 @@
+/**
+ * @flow
+ * @file Suggested Collaborator utility functions
+ * @author Box
+ */
+
+import {
+    MAX_GAPS_FUZZY_MATCH,
+    MAX_SUGGESTIONS_TO_SHOW,
+    MIN_CHARACTERS_FOR_MATCHING,
+    SUGGESTED_COLLAB_CONTACT_TYPE,
+} from '../constants';
+import fuzzySearch from './fuzzySearch';
+
+/**
+ * Function to convert a suggested collaborator object to a contact object
+ *
+ * @param {SuggestedCollab} suggestedCollab The suggested collaborator object to convert
+ * @returns {Contact} The converted contact object
+ */
+function convertSuggestedCollabToContact(suggestedCollab: SuggestedCollab): Contact {
+    return {
+        email: suggestedCollab.email,
+        id: suggestedCollab.id,
+        name: suggestedCollab.name,
+        text: suggestedCollab.name,
+        type: SUGGESTED_COLLAB_CONTACT_TYPE,
+        value: suggestedCollab.email,
+    };
+}
+
+/**
+ * Function to compute suggested collaborators given a list of contacts and cached suggested collaborators.
+ * Used by input components to help augment API results with cached suggestions using a fuzzy search.
+ *
+ * @param {Array<Contact>} contacts Contacts returned from the server
+ * @param {SuggestedCollabLookup} suggestedCollaborators Cached suggested collab lookup to match on
+ * @param {string} inputValue Value the user typed into the input box searching for collabs
+ * @param {number} maxSuggestions Maximum number of items to suggest/return, default 3
+ * @param {number} minCharacters Minimum number of characters before searching, default 3
+ * @returns {Array<Contact>} Suggested collaborators formatted as Contacts in recommended order
+ */
+function computeSuggestedContacts(
+    contacts: Array<Contact>,
```

Comment:
One thing I wanted to point out is the data format here. A different implementation of the PillSelector use a data type like `Array<{ text: string, value: string, data: Object }>`. Where `data` is similar to `Contact`. I think this is fine because you can just pass the data field to this function or you can convert one data type to another but I think it's worth pointing out because the pill selector itself might not be happy about the `Contact` format that is returned. If the other implementation of PillSelector is going to use this you might need to add another function and param to create the data format it wants or you can refactor the other implementation to use this data format. Not a huge deal, but I think it's worth pointing out up-front.

Reply:
Totally agree, there are a few other things like that such as whether the consumers should pass in the constants etc. I decided to keep it simple and didn't want to go too deep on restructuring all the types and such until we add the next use case(s).